core-bundle/src/Resources/contao/library/Contao/Environment.php

```diff
@@ -296,9 +296,9 @@ protected static function httpHost()
 		}
 		else
 		{
-			$host = $_SERVER['SERVER_NAME'];
+			$host = $_SERVER['SERVER_NAME'] ?? '';
 
-			if ($_SERVER['SERVER_PORT'] != 80)
+			if (isset($_SERVER['SERVER_PORT']) && $_SERVER['SERVER_PORT'] != 80)
```

Comment:
Our PHP8 adjustments always try to mimic the previous behaviour, always cast to `null`

```suggestion
			if (($_SERVER['SERVER_PORT'] ?? null) != 80)
```